### PR TITLE
htop: update to 3.0.0-1

### PR DESCRIPTION
**Pull request by graysky2**
new upstream, new release taken from ChangeLog<sup>1</sup>:
> New maintainers - after a prolonged period of inactivity from Hisham,  the creator and original maintainer, a team of community maintainers  have volunteered to take over a fork at https://htop.dev and https://github.com/htop-dev to keep the project going.

1. https://github.com/htop-dev/htop/blob/master/ChangeLog

Signed-off-by: graysky <graysky@archlinux.us>

Maintainer: @neheb @hnyman @champtar 
Compile tested: x86_64
Run tested: ipq806x, R7800, latest git

### CODE REVIEW

**Comment by graysky2** [comment]
...not sure why this failed in buildpr.
```
% make -j9 package/htop/compile  
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/scratch/union/scripts/config'
 make[1] package/htop/compile
 make[2] -C package/libs/toolchain compile
 make[2] -C package/libs/ncurses host-compile
 make[2] -C package/libs/zlib compile
 make[2] -C package/libs/ncurses compile
 make[2] -C feeds/packages/admin/htop compile
make -j9 package/htop/compile  6.64s user 1.29s system 104% cpu 7.565 total

% echo $?
0
```

**Comment by hnyman** [comment]
> not sure why this failed in buildpr.

The buildpr log shows that the error is about your name (as just an username is unacceptable):

```
=== Checking commit 'b8ac6ea5752c5f30af131aaeab9ec816c8b9fbea'
Author name (graysky) need to be your real name 'firstname lastname'
Commit subject line seems ok (htop: update to 3.0.0-1)
Signed-off-by match author

Exited with code exit status 1

CircleCI received exit code 1

```

**Comment by hnyman** [comment]
Is there any acknowledgement by the original site that this new fork is the legitimate successor?

**Comment by graysky2** [comment]
@hnyman - Thank you for pointing out the username issue.  I neglected to make the change globally and I just forked this repo this AM. Rebased and force pushed.

As to your question, I have not found an official announcement from Hisham, but I can tell you that several distros including Debian1, Arch2 and Gentoo3 have made this switch recently.

1. https://packages.debian.org/sid/htop
2. https://github.com/archlinux/svntogit-packages/commit/a2612366fc5637ea0d9c2120d418e9684eeedd25#diff-8d0411b338c83cd8cd8ad9d9db127101
3. https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=f8e2351e2fc6d869593ce5f1775526ec399270bf

**Comment by hnyman** [comment]
Good enough reasoning for me. (I have wondered about the stalled development...)

@champtar 
Any objections?

File: admin/htop/Makefile [hunk 1]
```diff
@@ -8,17 +8,19 @@
...
 PKG_NAME:=htop
-PKG_VERSION:=2.2.0
-PKG_RELEASE:=3
+PKG_VERSION:=3.0.0
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://hisham.hm/htop/releases/$(PKG_VERSION)/
-PKG_HASH:=d9d6826f10ce3887950d709b53ee1d8c1849a70fa38e91d5896ad8cbc6ba3c57
+PKG_SOURCE_URL:=https://github.com/htop-dev/htop/archive/$(PKG_VERSION)/
```
Comment (neheb) [inline admin/htop/Makefile:15]: use codeload for the URL instead:

```
https://codeload.github.com/htop-dev/htop/tar.gz/$(PKG_VERSION)?
```

File: admin/htop/Makefile [hunk 2]
```diff
@@ -8,17 +8,19 @@
...
+PKG_HASH:=1c0661f0ae5f4e2874da250b60cd515e4ac4c041583221adfe95f10e18d1a4e6
 
 PKG_LICENSE:=GPL-2.0
```
Comment (neheb) [inline admin/htop/Makefile:18]: Should be -or-later.

File: admin/htop/Makefile (hunk 1 above)
Comment (graysky2) [reply admin/htop/Makefile:15]: I get 403 invalid when I use the codeland. code you suggested.

File: admin/htop/Makefile (hunk 2 above)
Comment (graysky2) [reply admin/htop/Makefile:18]: I can change
```
-PKG_LICENSE:=GPL-2.0
+PKG_LICENSE:=GPL-2.0-or-later
```

File: admin/htop/Makefile (hunk 1 above)
Comment (graysky2) [reply admin/htop/Makefile:15]: Never tried codeland.... Is the literal ? at the end intended to be there or should it be a forward slash?

Comment (neheb) [reply admin/htop/Makefile]: Renames the file. That URL generates 3.0.0.tar.gz , which can conflict with other packages.

The typical way to fix this is to use ? and let the build system rename it.

Another way to fix this is to use curl with the -J parameter. I haven't seen anyone wanting to do this.

File: admin/htop/Makefile (hunk 2 above)
Comment (neheb) [reply admin/htop/Makefile]: Yes.

Comment (graysky2) [reply admin/htop/Makefile]: Done, thanks.

**Pull request merged**